### ISSUE TITLE
[Global] Missing focus ring/border in Windows High Contrast

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#1251]
## Issue summary

Several UI elements are missing focus rings and/or borders when navigating in Windows High Contrast mode.

- Country/lang missing focus ring
- Submit controls missing border and focus ring
- No focus ring on inputs
- "Show now", "View all" controls
- Featured product select and cta controls do not feature focus ring
- Filter (remove) controls do not feature focus ring
- Product image controls do not feature focus ring
- Product buy/add to cart controls do not feature focus ring or border
- Product model "View my cart"/checkout controls do not feature focus ring or border
- Cart "remove" and checkout controls do not feature focus ring
- Cart "your cart is empty" control does not feature focus ring

#### CSS

```css
.field__input:focus,
.select__select:focus,
.customer .field input:focus,
.customer select:focus,
.localization-form__select:focus {
  /* … */
  outline: transparent;
}

.button:focus-visible,
.button:focus,
.button.focused,
.shopify-payment-button__button--unbranded:focus-visible,
.shopify-payment-button [role='button']:focus-visible,
.shopify-payment-button__button--unbranded:focus,
.shopify-payment-button [role='button']:focus {
  outline: 0;
  /* … */
}

.button, .shopify-challenge__button, .customer button {
  /* … */
  border: 0;
  /* … */
}

.customer .field input:focus, .customer select:focus, .field__input:focus, .localization-form__select:focus, .select__select:focus {
  /* … */
  outline: transparent;
  /* … */
}

a.active-facets__button:focus-visible {
  outline: none;
  /* … */
}

.product__media-toggle:focus-visible {
  outline: 0;
  /* … */
}

.thumbnail[aria-current]:focus, .thumbnail.focused {
  outline: 0;
  /* … */
}
```

## Steps to reproduce

1. Go to a demo store with Edge browser in Windows
2. With an active High Contrast theme, <kbd>Tab</kbd> through and review focus rings and borders of interactive controls

### Expected

- Focus rings and borders to be visible in High Contrast theme.

### Actual

- Focus rings and borders are not visible.

## Recommendation

For applicable CSS definitions:

1. Adjust `outline: none;` or `outline: 0;` to `outline: transparent solid 1px;`
2. Adjust `outline: transparent;` to `outline: transparent solid 1px;`
3. Adjust `border: 0;` to `border: transparent solid 1px;`

These changes will provide a focus ring/border to the appripriate element in Windows High Contrast themes only.

#### CSS

```css
.field__input:focus,
.select__select:focus,
.customer .field input:focus,
.customer select:focus,
.localization-form__select:focus {
  /* … */
  outline: transparent solid 1px;
}

.button:focus-visible,
.button:focus,
.button.focused,
.shopify-payment-button__button--unbranded:focus-visible,
.shopify-payment-button [role='button']:focus-visible,
.shopify-payment-button__button--unbranded:focus,
.shopify-payment-button [role='button']:focus {
  outline: solid transparent 1px;
  /* … */
}

.button, .shopify-challenge__button, .customer button {
  /* … */
  border: transparent solid 1px;
  /* … */
}

.customer .field input:focus, .customer select:focus, .field__input:focus, .localization-form__select:focus, .select__select:focus {
  /* … */
  outline: transparent solid 1px;
  /* … */
}
a.active-facets__button:focus-visible {
  outline: transparent solid 1px;
  /* … */
}

.product__media-toggle:focus-visible {
  outline: transparent solid 1px;
  /* … */
}

.thumbnail[aria-current]:focus, .thumbnail.focused {
  outline: transparent solid 1px;
  /* … */
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Dawn version:** 2.5.0
- **Browser:** Any
- **Component:** Global
- **WCAG Principle:** Perceivable
- **WCAG SC:** [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
- **Severity:** High
- **Effort:** Low
